pkg/client/clients_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package client
+
+import (
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+
+	"github.com/apache/yunikorn-core/pkg/common"
+	"github.com/apache/yunikorn-k8shim/pkg/common/test"
+)
+
+const (
+	noOfInformers = 9
+)
+
+func TestWaitForSync(t *testing.T) {
```

Comment:
Maybe we can do a bit refactor to `Clients` to simplify the test. 

1. add a new method `informers` to  `Clients` to return all informers
2. rewrite `WaitForSync` and `Run` by looping all informers
3. in this test we can create a mock clients to override the new method `informers` to return a single mock informer

WDYT?

Reply:
Could we do it a separate ticket? In this one I'd rather just focus on the new tests.

Reply:
ok, I have filed https://issues.apache.org/jira/browse/YUNIKORN-2625